src/main/java/assignment/main/Game.java

```diff
@@ -144,38 +163,47 @@ public static void main(String[] args) {
 
         System.out.println("How many players? (2-8 players)");
         int playerCount = input.nextInt();
+        System.out.println("How many teams? (Enter a number less than 2 to turn teams off)");
+        int numOfTeams = input.nextInt();
         System.out.println("How large is the map? (5-50 for 2-4 players, 8-50 for 5-8 players)");
         int mapSize = input.nextInt();
 
-        startGame(playerCount, mapSize);
+        startGame(playerCount, numOfTeams, mapSize);
         generateHTMLFiles();
 
         boolean win = false;
         boolean[] winners = new boolean[playerCount];
 
         do {
-            for(int i = 0; i < playerCount; i++){
-                Player player = players[i];
-                boolean valid = false;
-                while(!valid){
-                    valid = menu(player, i);
-                }
-
-                if(map.getTileType(player.getPosition()) == TileType.TREASURE){
-                    win = true;
-                    winners[i] = true;
-                } else if (map.getTileType(player.getPosition()) == TileType.WATER){
-                    System.out.println("objects.Player " + (i+1) + " drowned!");
-                    player.resetToInitialPosition();
+            for(int i = 0; i < teams.length; i++) {
+                int previousTeamUpper = teams[0].getTeamSize()*i; // E.g: for 5 player teams; when team 2 turn, ptu = 5
```

Comment:
Are you sure this works for when number of players is not perfectly divisible by number of teams?